python/ambassador/ir/irhealthchecks.py

```diff
@@ -0,0 +1,256 @@
+                                f"IRHealthChecks: expected_statuses: status range start value {minCode} cannot be higher than the end {maxCode} for range. Ignoring expected status for health-check {hc}",
+                                log_level=logging.ERROR,
+                            )
+                            continue
+
+                        newRange = {"start": minCode, "end": maxCode}
+                        validStatuses.append(newRange)
+                    if len(validStatuses) > 0:
+                        http_mapper["expected_statuses"] = validStatuses
+                # Add the http health check to the config
+                mapper["http_health_check"] = http_mapper
+
+            # Process a gRPC health check
+            if grpc_health_check is not None:
+                if not isinstance(grpc_health_check, dict):
+                    self.post_error(
+                        f"IRHealthChecks: health_check.grpc: field must be an object, found {grpc_health_check}, Ignoring...",
+                        log_level=logging.ERROR,
+                    )
+                    continue
+
+                upstream_name = grpc_health_check.get("upstream_name", None)
+                if upstream_name is None:
+                    self.post_error(
+                        f"IRHealthChecks: health_check.grpc: required field upstream_name field not set, ignoring health-check {hc}",
+                        log_level=logging.ERROR,
+                    )
+                    continue
+                else:
+                    grpc_mapper: Dict[str, str] = {"service_name": upstream_name}
+
+                authority = grpc_health_check.get("authority", None)
+                if authority is not None:
+                    grpc_mapper["authority"] = authority
+
+                # Add the gRPC health check to the config
+                mapper["grpc_health_check"] = grpc_mapper
+            all_mappers.append(mapper)
+
+        # If nothing could be parsed successfully, post an error.
+        if len(all_mappers) == 0:
+            self.post_error(
+                f"IRHealthChecks: no valid health check could be parsed for config: {self._ir_config}",
+                log_level=logging.ERROR,
+            )
+            return None
+        return all_mappers
+
+    @staticmethod
+    def generate_headers_to_add(header_dict: dict) -> List[dict]:
+        headers = []
+        for k, v in header_dict.items():
+            append = True
+            if isinstance(v, dict):
+                if "append" in v:
+                    append = bool(v["append"])
+                headers.append({"header": {"key": k, "value": v["value"]}, "append": append})
+            else:
+                headers.append(
+                    {
+                        "header": {"key": k, "value": v},
+                        "append": append,  # Default append True, for backward compatability
+                    }
+                )
+        return headers
```

Comment:
This file is a lot of boilerplate for not very much functionality.

Just something to think about wrt improving the base `IRResource` in the future.

Reply:
Yeah, main driver was just that its a ton of checking and translating to jam into the v3cluster.py and there was already kind of a precedent with irerrorresponse.py although that one does do more than this file does.